cookbook/profiler/data_collector.rst

```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
 
     interface DataCollectorInterface
     {
-        /**
-         * Collects data for the given Request and Response.
-         *
-         * @param Request    $request   A Request instance
-         * @param Response   $response  A Response instance
-         * @param \Exception $exception An Exception instance
-         */
         function collect(Request $request, Response $response, \Exception $exception = null);
-
-        /**
-         * Returns the name of the collector.
-         *
-         * @return string The collector name
-         */
         function getName();
     }
 
-The value returned by ``getName()`` must be unique in the application. This value
-is also used to access the information later on (see :doc:`/cookbook/testing/profiling`
-for instance).
-
-The ``collect()`` method is responsible for storing the collected data in local
-properties.
+The
+:method:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollectorInterface::getName`
+method returns the name of the data collector and must be unique in the
+application. This value is also used to access the information later on (see
+:doc:`/cookbook/testing/profiling` for instance).
 
-.. caution::
-
-    As the profiler serializes data collector instances, you should not
-    store objects that cannot be serialized (like PDO objects), or you need
-    to provide your own ``serialize()`` method.
+The
+:method:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollectorInterface::collect`
+method is responsible for storing the collected data in local properties.
 
 Most of the time, it is convenient to extend
 :class:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollector` and
 populate the ``$this->data`` property (it takes care of serializing the
-``$this->data`` property)::
+``$this->data`` property). Imagine you create a new data collector that
+collects the method and accepted content types from the request::
+
+    // src/AppBundle/DataCollector/RequestCollector.php
+    namespace AppBundle\DataCollector;
 
-    // src/AppBundle/DataCollector/MyCollector.php
     use Symfony\Component\HttpKernel\DataCollector\DataCollector;
 
-    class MyCollector extends DataCollector
+    class RequestCollector extends DataCollector
     {
         public function collect(Request $request, Response $response, \Exception $exception = null)
         {
             $this->data = array(
-                'variable' => 'value',
+                'method' => $request->getMethod(),
+                'acceptable_content_types' => $request->getAcceptableContentTypes(),
```

Comment:
decided to go for a more real usage, to show a way to collect data and how to access collected data in the template. On top of that, using real use-case examples, things are often easier to understand.